packages/back-end/src/api/members/updateMemberRole.ts

```diff
@@ -88,18 +88,14 @@ export const updateMemberRole = createApiRequestHandler(
 
     const { member } = req.body;
 
-    const updatedEnvironments = member.environments
-      ? member.environments
-      : orgUser.environments;
-
-    const updatedLimitAccessByEnv =
-      updatedEnvironments.length > 0 ? true : false;
-
     const updatedMember: Member = {
       ...orgUser,
       role: member.role || orgUser.role,
-      environments: updatedEnvironments,
-      limitAccessByEnvironment: updatedLimitAccessByEnv,
+      environments: member.environments || orgUser.environments,
+      // If environments are passed in, recalculate limitAccessByEnvironment, otherwise, use existing value
+      limitAccessByEnvironment: member.environments
+        ? !!member.environments.length
+        : orgUser.limitAccessByEnvironment,
```

Comment:
Original role:
```ts
{
  role: "engineer",
  environments: ["dev", "production"],
  limitAccessByEnvironment: true
}
```

API Request:
```ts
{
  role: "admin"
}
```

Right now, I believe this PR will just throw an error, which is fine, but it might be better to just always set `limitAccessByEnvironment` to `!!member.environments?.length` instead of having that ternary.